### PR TITLE
Create Music Album and Genre classes

### LINKED ISSUE #20: [1pt] Create MusicAlbum class in a separate .rb file.
Create MusicAlbum class in a separate .rb file.

### LINKED ISSUE #21: [1pt] Create Genre class with an association to the Item class (in a separate .rb file).
Create Genre class with an association to the Item class (in a separate .rb file).

### LINKED ISSUE #22: [1pt] All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.
All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #23: [1pt] All Genre class properties visible in the diagram should be defined and set up in the constructor method.
All Genre class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #24: [1pt] Implement methods
1) add_item method in the Genre class

- should take an instance of the Item class as an input
- should add the input item to the collection of items
- should add self as a property of the item object (by using the correct setter from the item object)

2) can_be_archived?() in the MusicAlbum class

- should override the method from the parent class
- should return true if parent's method returns true AND if on_spotify equals true
- otherwise, it should return false

### PR DESCRIPTION
**Pull request by katarighe**
In this pull request, I have completed the following tasks: 

- Create `MusicAlbum` class in a separate `.rb` file.
- Create a `Genre` class with an association to the Item class (in a separate `.rb` file).
- All `MusicAlbum` class properties visible in the diagram are defined and set up in the constructor method.
- All `Genre` class properties visible in the diagram are defined and set up in the constructor method.

Apart from these tasks, I have also done implementing the methods as well:

1) `add_item` method in the Genre class

- should take an instance of the Item class as an input
- should add the input item to the collection of items
- should add self as a property of the item object (by using the correct setter from the item object)

2) `can_be_archived?()` in the `MusicAlbum` class

- should override the method from the parent class
- should return true if parent's method returns true AND if `on_spotify` equals true
- otherwise, it should return false

### CODE REVIEW

File: classes/genre.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require_relative 'item'
+
+class Genre
+  attr_accessor :id, :name, :items
```
Comment (jecode93) [inline classes/genre.rb]: According to the diagram, `id` and `items` must be private. You should use `attr_reader` for them!

File: classes/music_album.rb
Comment (jecode93) [inline classes/music_album.rb:1]: The method `can_be_archived?` is missing according to the diagram. Try to fix that and submit a again for another review.

File: classes/music_album.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require_relative 'item'
+
+class MusicAlbum < Item
+  attr_accessor :on_spotify
+
+  def initialize(on_spotify, archived, publish_date)
+    super(archived, publish_date)
```
Comment (jecode93) [inline classes/music_album.rb]: According to the diagram, the `archived` parameter is private, you should remove it from the constructor. You will need it in the `can_be_archived` method only.

**Review by jecode93 — CHANGES_REQUESTED** [review]
## Require changes 🙅🏽‍♂️ 

See comments below!

File: classes/genre.rb (hunk 1 above)
Comment (katarighe) [reply classes/genre.rb]: ✅ Done and corrected.

File: classes/music_album.rb
Comment (katarighe) [reply classes/music_album.rb:1]: ✅ Done and corrected.

**Review by jecode93 — APPROVED** [review]
# Approved

There is nothing else to say, then it's time to merge it!

**Pull request merged**